pkg/util/util.go

```diff
@@ -785,11 +787,46 @@ func GetGitHubZipURL(repoURL string) (string, error) {
 		repo = strings.TrimSuffix(repo, ".git")
 	}
 
-	// TODO: pass branch or tag from devfile
-	branch := "master"
+	// Extract reference checkout for project
+	references := make(map[string]string)
+	if project.Source.Branch != nil && *project.Source.Branch != "" {
+		references["Branch"] = *project.Source.Branch
+	}
+
+	if project.Source.CommitId != nil && *project.Source.CommitId != "" {
+		references["CommitId"] = *project.Source.CommitId
+	}
+
+	if project.Source.Tag != nil && *project.Source.Tag != "" {
+		references["Tag"] = *project.Source.Tag
+	}
+
+	if project.Source.StartPoint != nil && *project.Source.StartPoint != "" {
+		references["StartPoint"] = *project.Source.StartPoint
+	}
+
+	if len(references) > 1 {
```

Comment:
You could have a branch and commitID as per the devfiles, but a commit is universal in a repo which means that one commit can be in multiple branches. The github api accepts only one reference. Maybe we should not throw an error if branch and commitID and use the commitID as the reference. But we still throw an error in the other cases?

Reply:
Okay, that's fair. If github only accepts one value as a reference, then I think we should throw an error, to prevent any confusion on the user's end.